src/testing/core/__tests__/createTestSchema.test.tsx

```diff
@@ -1028,12 +1000,158 @@ describe("schema proxy", () => {
+            title: "A New Book",
+          },
+        },
+      });
+    }
+  });
+
+  it.only("resets the schema with schema.reset()", async () => {
+    const resetTestSchema = createTestSchema(schema, {
+      resolvers: {
+        Query: {
+          viewer: () => ({
+            book: {
+              text: "Hello World",
+              title: "Orlando: A Biography",
+            },
+          }),
+        },
+        Book: {
+          __resolveType: (obj) => {
+            if ("text" in obj) {
+              return "TextBook";
+            }
+            if ("colors" in obj) {
+              return "ColoringBook";
+            }
+            throw new Error("Could not resolve type");
+          },
+        },
+      },
+    });
+    const Profiler = createDefaultProfiler<ViewerQueryData>();
+
+    resetTestSchema.add({
+      resolvers: {
+        Query: {
+          viewer: () => ({
+            book: {
+              text: "Hello World",
+              title: "The Waves",
+            },
+          }),
+        },
+      },
+    });
+
+    using _fetch = createSchemaFetch(resetTestSchema).mockGlobal();
+
+    const client = new ApolloClient({
+      cache: new InMemoryCache(),
+      uri,
+    });
+
+    const query: TypedDocumentNode<ViewerQueryData> = gql`
+      query {
+        viewer {
+          id
+          name
+          age
+          book {
+            id
+            title
+            publishedAt
+            ... on ColoringBook {
+              colors
+            }
+          }
+        }
+      }
+    `;
+
+    const Fallback = () => {
+      return <div>Loading...</div>;
+    };
+
+    const App = () => {
+      return (
+        <React.Suspense fallback={<Fallback />}>
+          <Child />
+        </React.Suspense>
+      );
+    };
+
+    const Child = () => {
+      const result = useSuspenseQuery(query);
+      console.log(result.data);
+      Profiler.mergeSnapshot({
+        result,
+      } as Partial<{}>);
+
+      return (
+        <div>
+          Hello<button onClick={() => result.refetch()}>Refetch</button>
+        </div>
+      );
+    };
+
+    renderWithClient(<App />, {
+      client,
+      wrapper: Profiler,
+    });
+
+    // initial suspended render
+    await Profiler.takeRender();
+
+    {
+      const { snapshot } = await Profiler.takeRender();
+
+      expect(snapshot.result?.data).toEqual({
+        viewer: {
+          __typename: "User",
+          age: 42,
+          id: "1",
+          name: "String",
+          book: {
+            __typename: "TextBook",
+            id: "1",
+            publishedAt: "2024-01-01",
+            // value set in this test with .add
+            title: "The Waves",
           },
         },
       });
     }
 
-    unmount();
+    resetTestSchema.reset();
+
+    const user = userEvent.setup();
+
+    await act(() => user.click(screen.getByText("Refetch")));
+
+    // initial suspended render
+    await Profiler.takeRender();
+
+    {
```

Comment:
It just doesn't suspend here after clicking refetch - probably because the fetch doesn't take long enough that React tries to suspend.
```suggestion
    await act(() => user.click(screen.getByText("Refetch")));

    {
```

```suggestion
    await act(() => user.click(screen.getByText("Refetch")));

    {
```

Reply:
After merging in the other PRs this is now suspending on refetch and passing 👍